security/fma-interop-portal.md

```diff
@@ -0,0 +1,212 @@
+# FMA: Interop Support in OptimismPortal
+
+|---------------------|----------------|
+| Author              | Kelvin Fichter |
+| Created at          | 2025-04-03     |
+| Needs Approval From |                |
+| Other Reviewers     |                |
+| Status              | Draft          |
+
+## Introduction
+
+This Failure Mode Analysis discusses changes made to the `OptimismPortal` and a number of
+surrounding contracts to support interop between two OP Stack chains.
+
+### AnchorStateRegistry as source of truth
+
+The `OptimismPortal` is updated to use the `AnchorStateRegistry` as the source of truth for the
+validity of Dispute Game instances. This means that the `AnchorStateRegistry` now decides whether a
+given Dispute Game is valid and holds important state like the retirement timestamp and the
+respected game type.
+
+## ETHLockbox for every OptimismPortal
+
+The `OptimismPortal` is updated to use the `ETHLockbox` contract to store ETH. The `ETHLockbox`
+supports multiple `OptimismPortal` contracts and has a function that allows an admin account to
+allow new portals to use the lockbox and to migrate funds from one lockbox to another. Migration
+functions in the `OptimismPortal` and the `ETHLockbox` and the overall migration process defined by
+these functions are in scope for this contest.
+
+## OptimismPortal proving function for Super Roots
+
+The `OptimismPortal` is updated to have a new version of `proveWithdrawalTransaction` that uses the
+Super Root proving path. The Super Root proving path is toggled by the variable `superRootsActive`
+which is meant to be triggered when a chain migrates to using Super Roots.
+
+## Failure Modes
+
+### FM1: Incorrect State Reporting by `AnchorStateRegistry`
+
+#### Description
+
+A bug within the `AnchorStateRegistry` causes it to return incorrect data about a dispute game
+(e.g., resolves incorrectly, reports wrong `respectedGameType`, incorrect `retirementTimestamp`).
+Since the `OptimismPortal` now relies entirely on the `AnchorStateRegistry` as the source of truth
+for game validity and state, incorrect data leads directly to `OptimismPortal` malfunction.
+
+#### Risk Assessment
+
+- Impact: CRITICAL
+  - Reasoning: Could lead to a **Withdrawal Safety Failure** or **Withdrawal Liveness Failure**.
+- Likelihood: LOW
+  - Reasoning: Assumes the `AnchorStateRegistry` is well-tested and audited as a core component.
+
+#### Mitigations
+
+- Rigorous testing (unit, integration, fuzzing) of `AnchorStateRegistry`.
+- Multiple independent audits of `AnchorStateRegistry`.
+
+#### Detection
+
+- Monitoring already exists for the correctness and age of the Anchor State.
+- Monitoring already exists for the correctness of resolved dispute games.
```

Comment:
- @smartcontracts: update monitoring to take the systemconfig or portal as input, so that when the ASR is updated it is automatically discovered rather than needing to manually update the ASR address. 
  - Same should be done for any monitoring which takes the DGF as input.